api/datastore/sql/sql.go

```diff
@@ -365,33 +370,57 @@ func (ds *sqlStore) RemoveApp(ctx context.Context, appName string) error {
 		}
 
 		deletes := []string{
-			`DELETE FROM logs WHERE app_name=?`,
-			`DELETE FROM calls WHERE app_name=?`,
-			`DELETE FROM routes WHERE app_name=?`,
+			`DELETE FROM logs WHERE app_id=?`,
+			`DELETE FROM calls WHERE app_id=?`,
+			`DELETE FROM routes WHERE app_id=?`,
 		}
-
 		for _, stmt := range deletes {
-			_, err := tx.ExecContext(ctx, tx.Rebind(stmt), appName)
+			_, err := tx.ExecContext(ctx, tx.Rebind(stmt), app.ID)
 			if err != nil {
 				return err
 			}
 		}
+
 		return nil
 	})
 }
 
-func (ds *sqlStore) GetApp(ctx context.Context, name string) (*models.App, error) {
-	query := ds.db.Rebind(`SELECT name, config, created_at, updated_at FROM apps WHERE name=?`)
-	row := ds.db.QueryRowxContext(ctx, query, name)
+func getAppTx(tx *sqlx.Tx, ctx context.Context, app string, isAppID bool, a *models.App) error {
```

Comment:
we shouldn't have to do this, it's also very error prone, as noted earlier in discussion about cache/hybrid endpoints. we only need 1 db method that looks up an app id for an app name, everything else should all be ids I think or we have to start thinking about weird race conditions